src/pages/Index.js

```diff
@@ -66,53 +123,159 @@ modalWithImage.setEventListeners();
+      editModal.renderLoading(false);
+    });
 }
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    settings,
-    renderer: renderCard,
-  },
+/*                                       */
+/*              avatar                   */
+/*                                       */
+
+function handleAvatarSubmit(data) {
+  editAvatarModal.renderLoading(true);
+  api
+    .updateAvatar({ avatar: data.link })
+    .then((res) => {
+      userInfo.setAvatar(res.avatar);
+      editAvatarModal.close();
+    })
+    .catch((err) => {
+      console.error("Failed to update user avatar:", err);
+    })
+    .finally(() => {
+      editAvatarModal.renderLoading(false);
+    });
+}
 
-  ".cards__list"
+const editAvatarModal = new ModalWithForm(
+  "#edit-avatar-modal",
+  handleAvatarSubmit
 );
+editAvatarModal.setEventListeners();
 
-cardSection.renderItems();
+profileAvatarContainer.addEventListener("click", () => {
+  editAvatarModal.open();
+});
 
 /*                                       */
-/*             Event Handlers            */
+/*          LIKE & DISLIKE               */
 /*                                       */
 
+function handleLikeClick(card) {
+  if (card.getIsLiked()) {
+    api
+      .removeLike(card.getCardId())
+      .then(() => {
+        card.setIsLiked(false);
+      })
+      .catch(console.error);
+  } else {
+    api
+      .addLike(card.getCardId())
+      .then(() => {
+        card.setIsLiked(true);
+      })
+      .catch(console.error);
+  }
+}
+
 function handleImageClick(name, link) {
   modalWithImage.open(name, link);
 }
 
-const user = new UserInfo("#modal-title-input", "#modal-description-input");
+/*                                       */
+/*              ADD CARD                 */
+/*                                       */
 
-function handleProfileEditSubmit(data) {
-  userInfo.setUserInfo({ name: data.title, description: data.description });
-  editModal.close();
+function handleAddCardSubmit(data) {
+  const name = data.title;
+  const link = data.link;
+  cardModal.renderLoading(true);
+  api
+    .createNewCard({ name, link })
+    .then((res) => {
+      const cardElement = createCard(res);
+      cardSection.addItem(cardElement);
+      cardModal.close();
+      addCardForm.reset();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      cardModal.renderLoading(false);
+    });
 }
 
-function handleAddCardSubmit(userInfo) {
-  const name = userInfo.title;
-  const link = userInfo.link;
-  const newCard = renderCard({ name, link });
-  cardSection.addItem(newCard);
-  cardModal.close();
+/*                                       */
+/*            DELETE CARD                */
+/*                                       */
+
+const confirmDeleteModal = new ModalWithConfirm("#delete-modal");
+confirmDeleteModal.setEventListeners();
+
+function handleDeleteClick(card) {
+  confirmDeleteModal.open();
+  confirmDeleteModal.renderLoading(true);
```

Comment:
Needs correction: you should start the loading right before the api request on line 263